kotlin-generator/src/android-client.ts

```diff
@@ -0,0 +1,153 @@
+import { AstRoot } from "@sdkgen/parser";
+import { generateClass, generateEnum, generateErrorClass, generateJsonAddRepresentation, generateKotlinTypeName, mangle } from "./helpers";
+
+interface Options {}
+
+export function generateAndroidClientSource(ast: AstRoot, options: Options) {
+    let code = `import android.content.Context
+import android.util.Base64
+import com.google.gson.*
+import com.google.gson.reflect.TypeToken
+import io.sdkgen.runtime.SdkgenHttpClient
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.async
+import java.lang.reflect.Type
+import java.text.SimpleDateFormat
+import java.util.*
+
+inline fun <reified T> Gson.fromJson(json: String) =
+    fromJson<T>(json, object : TypeToken<T>() {}.type)
+
+inline fun <reified T> Gson.fromJson(json: JsonElement) =
+    fromJson<T>(json, object : TypeToken<T>() {}.type)
+
+@ExperimentalCoroutinesApi
+@Suppress("DeferredIsResult", "unused")
+class ApiClient(
+    baseUrl: String,
+    applicationContext: Context,
+    defaultTimeoutMillis: Long = 10000L
+) : SdkgenHttpClient(baseUrl, applicationContext, defaultTimeoutMillis) {`;
+
+    code += `
+    private class ByteArrayDeserializer : JsonDeserializer<ByteArray> {
+        @Throws(JsonParseException::class)
+        override fun deserialize(
+            element: JsonElement,
+            arg1: Type?,
+            arg2: JsonDeserializationContext?
+        ): ByteArray {
+            try {
+                return Base64.decode(element.asString, Base64.DEFAULT)
+            } catch (e: Exception) {
+                throw JsonIOException(e.message)
+            }
+        }
+    }
+
+    private class CalendarDeserializer : JsonDeserializer<Calendar> {
+        @Throws(JsonParseException::class)
+        override fun deserialize(
+            element: JsonElement,
+            arg1: Type?,
+            arg2: JsonDeserializationContext?
+        ): Calendar {
+            val date = element.asString
+            val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS", Locale.US)
+
+            formatter.timeZone = TimeZone.getTimeZone("UTC")
+
+            try {
+                return Calendar.getInstance().apply { time = formatter.parse(date)!! }
+            } catch (e: Exception) {
+                throw JsonIOException(e.message)
+            }
+        }
+    }
+
+    private val gson = GsonBuilder()
+        .registerTypeAdapter(object : TypeToken<Calendar>() {}.type, CalendarDeserializer())
```

Comment:
How will it differ a `date` from a `datetime`?

Reply:
Probably it needs to generate an annotation on these fields to use one specific adapter for each case: https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.0/com/google/gson/annotations/JsonAdapter.html